parser_library/src/context/hlasm_context.cpp

```diff
@@ -160,7 +160,8 @@ void hlasm_context::add_global_system_vars()
     auto SYSDATC = ids().add("SYSDATC");
     auto SYSDATE = ids().add("SYSDATE");
     auto SYSTIME = ids().add("SYSTIME");
-    auto SYSPARM = ids().add("SYSPARM");
+    auto SYSPARM = asm_options_.find("SYSPARM") == asm_options_.end() ? ids().add("SYSPARM")
```

Comment:
This seems wrong. I suspect that what this code does is just getting the name "SYSPARM" for the variable and not the value yet. I think that happens at `auto val = std::make_shared<set_symbol<C_t>>(SYSPARM, true, true);`

Reply:
It will try to find the value for the key "SYSPARM" and if not found will simply add "SYSPARM" else the value found in asm_options

Reply:
But the `SYSPARM` option is always called `SYSPARM` and as far as I can tell that's what this code is supposed to do.